test/common/test_common.cpp

```diff
@@ -253,39 +274,298 @@ TEST (PCL, PointCloud)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  cloud.transient_erase (cloud.end () - 1);
+  EXPECT_TRUE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_EQ (cloud.size(), (640*480) - 1);
+}
+
+TEST_F (pointCloudTest, transient_erase_with_iterator)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  cloud.transient_erase (cloud.begin (), cloud.end ());
+  EXPECT_TRUE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_EQ (cloud.size(), 0);
+}
+
+TEST_F (pointCloudTest, unorganized_concatenate)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  PointCloud<PointXYZ> new_unorganized_cloud;
+  PointCloud<PointXYZ>::concatenate (new_unorganized_cloud, cloud);
+  EXPECT_FALSE (new_unorganized_cloud.isOrganized ());
+  EXPECT_EQ (new_unorganized_cloud.width, 640*480);
+}
+
+TEST_F (pointCloudTest, unorganized_concatenate_with_argument_return)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  PointCloud<PointXYZ> new_unorganized_cloud;
+  PointCloud<PointXYZ>::concatenate (new_unorganized_cloud, cloud);
+  PointCloud<PointXYZ> unorganized_cloud_out;
+  PointCloud<PointXYZ>::concatenate (new_unorganized_cloud, cloud, unorganized_cloud_out);
+  EXPECT_FALSE (unorganized_cloud_out.isOrganized ());
+  EXPECT_EQ (unorganized_cloud_out.width, 640*480*2);
+}
+
+TEST_F (pointCloudTest, unorganized_concatenate_with_assignment_return)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  PointCloud<PointXYZ> unorganized_cloud;
+  PointCloud<PointXYZ>::concatenate (unorganized_cloud, cloud);
+  PointCloud<PointXYZ> unorganized_cloud_out = cloud + unorganized_cloud;
+  EXPECT_FALSE (unorganized_cloud_out.isOrganized ());
+  EXPECT_EQ (unorganized_cloud_out.width, 640*480*2);
+}
+
+TEST_F (pointCloudTest, unorganized_concatenate_with_plus_operator)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  PointCloud<PointXYZ> unorganized_cloud;
+  unorganized_cloud += cloud;
+  EXPECT_FALSE (unorganized_cloud.isOrganized ());
+  EXPECT_EQ (unorganized_cloud.width, 640*480);
+}
+
+TEST_F (pointCloudTest, at_with_throw)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  PointCloud<PointXYZ> unorganized_cloud;
+  unorganized_cloud += cloud;
+  EXPECT_THROW({unorganized_cloud.at (5, 5);}, UnorganizedPointCloudException);
+}
+
+TEST_F (pointCloudTest, at_no_throw)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
+  const auto& point_at = cloud.at (cloud.width - 1, cloud.height - 1);
+  EXPECT_EQ(&point_at, &cloud.back());
+}
+
+TEST_F (pointCloudTest, organized_concatenate)
+{
+  cloud.resize (640, 480, PointXYZ (1, 1, 1));
+  PointCloud<PointXYZ> organized_cloud1 = cloud;
+  PointCloud<PointXYZ> organized_cloud2 = cloud;
+  EXPECT_TRUE (organized_cloud1.isOrganized ());
+  EXPECT_TRUE (organized_cloud2.isOrganized ());
+  PointCloud<PointXYZ> organized_cloud_out = organized_cloud1 + organized_cloud2;
+  EXPECT_FALSE (organized_cloud_out.isOrganized ());
+  EXPECT_EQ (organized_cloud_out.width, 614400);
```

Comment:
Better to write the formula: `organized_cloud1.size() + organized_cloud2.size()`

Reply:
Are you suggesting to replace EXPECT_EQ (organized_cloud_out.width, 614400); with EXPECT_EQ (organized_cloud1.size() + organized_cloud2.size(), 614400);

Reply:
```suggestion
  EXPECT_EQ (organized_cloud_out.width, organized_cloud1.size() + organized_cloud2.size());
```

Reply:
What you suggested doesn't seem to be kosher because we need to assert that organized_cloud1.size() + organized_cloud2.size() is correct (which we don't do anywhere), before we can use it as ground truth.

Reply:
Basically, we're saying that width of concatenated cloud is equal to sum of size of the 2 parent clouds